app/components/enter-hours.tsx

```diff
@@ -0,0 +1,298 @@
+"use client"
+
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card"
+import {
+    Popover,
+    PopoverContent,
+    PopoverTrigger,
+} from "@/components/ui/popover"
+import { ScrollArea } from "@radix-ui/react-scroll-area"
+import { zodResolver } from "@hookform/resolvers/zod"
+import { useForm } from "react-hook-form"
+import { Button } from "@/components/ui/button"
+import * as z from "zod"
+import {
+    Form,
+    FormControl,
+    FormField,
+    FormItem,
+    FormLabel,
+    FormMessage,
+} from "@/components/ui/form"
+import {
+    Select,
+    SelectContent,
+    SelectItem,
+    SelectTrigger,
+    SelectValue,
+} from "@/components/ui/select"
+import { useState } from "react"
+import { Calendar } from "@/components/ui/calendar"
+import { cn } from "@/lib/utils"
+import { format } from "date-fns"
+import { CalendarDays } from 'lucide-react';
+
+
+const enterHoursSchema = z.object({
```

Comment:
What I'd recommend is to make the `rulesetDateSchema` here exportable:

https://github.com/troypoulter/toil-calculator/blob/20bcab4be25d8e2a2c160e11441ec566f25cc62d/app/interfaces/Ruleset.ts#L7-L37

And then do something similar to what I did here:

https://github.com/troypoulter/toil-calculator/blob/20bcab4be25d8e2a2c160e11441ec566f25cc62d/app/interfaces/Ruleset.ts#L39-L61

Where I used the `and` to merge the two schemas together to generate one schema and then infer one type.

This way, you get the validation of the `startTime` and `endTime` of ensuring the `endTime` appears afterwards and avoid code duplication.